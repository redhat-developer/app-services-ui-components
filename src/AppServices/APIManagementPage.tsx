import {
  Button,
  ButtonVariant,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderMain,
  CardTitle,
  Grid,
  GridItem,
  List,
  ListItem,
  Stack,
  StackItem,
  Text,
  TextContent,
  Title,
  TitleSizes,
} from "@patternfly/react-core";
import type { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import { ApiManagementMechApiFullColorPink } from "../images";
import {
  MarketingPageHero,
  MarketingPageSection,
  MarketingPageVideoCard,
} from "./components";
import { ExternalLinkAltIcon } from "@patternfly/react-icons";

export const APIManagementPage: FunctionComponent = () => {
  const { t } = useTranslation();

  return (
    <>
      <MarketingPageHero
        title={t("apimgmtoverview:heroTitle")}
        tagLine={t("apimgmtoverview:heroTagline")}
        description={t("apimgmtoverview:heroDescription")}
        heroImage={ApiManagementMechApiFullColorPink}
        heroImageSize={478}
        heroImageCanRepeat={false}
        heroImagePositionY={-99}
      />
      <MarketingPageSection>
        <Grid hasGutter lg={6}>
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <CardTitle>
                  <Title headingLevel="h3">
                    {t("apimgmtoverview:alreadyHaveCardTitle")}
                  </Title>
                </CardTitle>
              </CardHeaderMain>
            </CardHeader>
            <CardBody>{t("apimgmtoverview:alreadyHaveCardMainText")}</CardBody>
            <CardFooter>
              <Stack hasGutter>
                <StackItem>
                  <Button
                    data-testid="cardHaveRHOAM-buttonGoOCM"
                    variant={ButtonVariant.secondary}
                    component="a"
                    href="https://cloud.redhat.com/openshift/"
                  >
                    {t("apimgmtoverview:alreadyHaveCardCallToActionButton")}
                  </Button>
                </StackItem>
                <StackItem>
                  <Button
                    data-testid="cardHaveRHOAM-linkViewDocs"
                    variant={ButtonVariant.link}
                    component="a"
                    href="https://access.redhat.com/products/red-hat-openshift-api-management"
                    target="_blank"
                  >
                    {t("apimgmtoverview:viewDocumentation")}{" "}
                    <ExternalLinkAltIcon className="pf-u-ml-sm" />
                  </Button>
                </StackItem>
              </Stack>{" "}
            </CardFooter>
          </Card>
          <Card>
            <CardHeader>
              <CardTitle>
                <Title headingLevel="h3">
                  {t("apimgmtoverview:wantToTryCardTitle")}
                </Title>
              </CardTitle>
            </CardHeader>
            <CardBody>{t("apimgmtoverview:wantToTryCardMainText")}</CardBody>
            <CardFooter>
              <Stack hasGutter>
                <StackItem>
                  <Button
                    data-testid="cardTryRHOAM-buttonGetStarted"
                    variant={ButtonVariant.secondary}
                    component="a"
                    href="https://developers.redhat.com/products/rhoam/getting-started"
                    target="_blank"
                  >
                    {t("apimgmtoverview:wantToTryCardCallToActionButton")}
                    <ExternalLinkAltIcon className="pf-u-ml-md" />
                  </Button>
                </StackItem>
                <StackItem>
                  <Button
                    data-testid="cardTryRHOAM-linkViewDocs"
                    variant={ButtonVariant.link}
                    component="a"
                    href="https://access.redhat.com/products/red-hat-openshift-api-management"
                    target="_blank"
                  >
                    {t("apimgmtoverview:viewDocumentation")}{" "}
                    <ExternalLinkAltIcon className="pf-u-ml-sm" />
                  </Button>
                </StackItem>
              </Stack>
            </CardFooter>
          </Card>
        </Grid>
      </MarketingPageSection>

      <MarketingPageSection variant="light">
        <Title size={TitleSizes.xl} headingLevel="h3" className="pf-u-mb-lg">
          {t("apimgmtoverview:videoSectionTitle")}
        </Title>
        <Grid hasGutter>
          <GridItem md={7}>
            <MarketingPageVideoCard
              src={"https://www.youtube.com/embed/NzNgc0f75pc"}
              title={t("apimgmtoverview:videoSectionTitle")}
            />
          </GridItem>
          <GridItem md={5}>
            <TextContent>
              <Text className="pf-u-color-200 pf-u-ml-md">
                {t("apimgmtoverview:videoSectionInThisVideo")}
              </Text>
              <List className="app-services-ui--icon-list">
                <ListItem>
                  {t("apimgmtoverview:videoSectionBulletBuild")}
                </ListItem>
                <ListItem>
                  {t("apimgmtoverview:videoSectionBulletImport")}
                </ListItem>
                <ListItem>
                  {t("apimgmtoverview:videoSectionBulletAdd")}
                </ListItem>
              </List>
            </TextContent>
          </GridItem>
        </Grid>
      </MarketingPageSection>
    </>
  );
};

export default APIManagementPage;
