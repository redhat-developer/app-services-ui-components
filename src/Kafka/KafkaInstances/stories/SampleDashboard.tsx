export default () => <div>Dashboard</div>;
