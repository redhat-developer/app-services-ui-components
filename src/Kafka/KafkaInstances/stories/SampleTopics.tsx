export default () => <div>Topics</div>;
