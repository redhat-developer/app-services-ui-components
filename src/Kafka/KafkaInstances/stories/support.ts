import { addHours } from "date-fns";
import { useEffect, useState } from "react";
import type { KafkaInstance } from "../../types";

const now = new Date().toISOString();
const future = addHours(new Date(), 19).toISOString();
const INSTANCES: KafkaInstance[] = [
  {
    id: "1",
    name: "foo",
    createdAt: now,
    updatedAt: now,
    expiryDate: undefined,
    owner: "owner",
    provider: "aws",
    region: "region",
    status: "provisioning",
    plan: "standard",
    size: "1",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
  {
    id: "2",
    name: "bar",
    createdAt: now,
    updatedAt: now,
    expiryDate: future,
    owner: "owner",
    provider: "aws",
    region: "region",
    status: "deleting",
    plan: "developer",
    size: "1",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
  {
    id: "3",
    name: "baz",
    createdAt: now,
    updatedAt: now,
    expiryDate: future,
    owner: "owner",
    provider: "azure",
    region: "region",
    status: "ready",
    plan: "standard",
    size: "2",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
];
export const useInstances = () => {
  const [instances, setInstances] = useState<KafkaInstance[]>();

  useEffect(() => {
    const timer = setTimeout(() => {
      setInstances(INSTANCES);
    }, 500);
    return () => clearTimeout(timer);
  }, []);
  return {
    instances,
    loading: instances === undefined,
  };
};
export const useInstance = (id: string) => {
  const { instances, loading } = useInstances();
  const instance = instances?.find((i) => i.id === id);

  return {
    instance,
    loading: loading || instance === undefined,
  };
};
