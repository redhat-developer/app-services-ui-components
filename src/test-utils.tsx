import type { RenderOptions, RenderResult } from "@testing-library/react";
import { act, render, waitFor } from "@testing-library/react";
import type { FunctionComponent, ReactElement } from "react";
import { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { I18nProvider } from "./I18n";
import { AlertProvider } from "../src/shared/Alerts";

const suspenseTestId = "i18n-suspense";

const AllTheProviders: FunctionComponent = ({ children }) => {
  return (
    <Router>
      <AlertProvider>
        <I18nProvider
          lng={"en"}
          resources={{
            en: {
              common: () => import("../locales/en/common.json"),
              "create-kafka-instance": () =>
                import("../locales/en/create-kafka-instance.json"),
              kafka: () => import("../locales/en/kafka.json"),
              metrics: () => import("../locales/en/metrics.json"),
              overview: () => import("../locales/en/overview.json"),
              datascienceoverview: () =>
                import("../locales/en/datascienceoverview.json"),
              kafkaoverview: () => import("../locales/en/kafkaoverview.json"),
              topic: () => import("../locales/en/topic.json"),
              apimgmtoverview: () =>
                import("../locales/en/apimgmtoverview.json"),
              "manage-kafka-permissions": () =>
                import("../locales/en/manage-kafka-permissions.json"),
              "create-topic": () => import("../locales/en/create-topic.json"),
              "message-browser": () =>
                import("../locales/en/message-browser.json"),
              "overview-v2": () => import("../locales/en/overview-v2.json"),
              "overview-v3": () => import("../locales/en/overview-v3.json"),
              "overview-v4": () => import("../locales/en/overview-v4.json"),
              "kafkaoverview-v2": () =>
                import("../locales/en/kafkaoverview-v2.json"),
              "kafkaoverview-v3": () =>
                import("../locales/en/kafkaoverview-v3.json"),
              "connection-tab": () =>
                import("../locales/en/connection-tab.json"),
              "service-registry": () =>
                import("../locales/en/service-registry.json"),
              "connection-tab-p1": () =>
                import("../locales/en/connection-tab-p1.json"),
            },
            it: {
              common: () => Promise.resolve({ delete: "Elimina" }),
            },
          }}
          debug={false}
        >
          <Suspense fallback={<div data-testid={suspenseTestId}>loading</div>}>
            {children}
          </Suspense>
        </I18nProvider>
      </AlertProvider>
    </Router>
  );
};

const AllTheProvidersWithRoot: FunctionComponent = ({ children }) => (
  <AllTheProviders>
    <div id={"root"}>{children}</div>
  </AllTheProviders>
);
const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const renderDialog = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProvidersWithRoot, ...options });

async function waitForI18n(r: RenderResult) {
  await waitFor(() => {
    expect(r.queryByTestId(suspenseTestId)).not.toBeInTheDocument();
  });
}

async function waitForPopper() {
  await act(async () => {
    /* let popper do its updates */
  });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderDialog, waitForI18n, waitForPopper };
