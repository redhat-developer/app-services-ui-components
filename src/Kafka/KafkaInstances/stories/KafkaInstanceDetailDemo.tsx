import {
  Breadcrumb,
  BreadcrumbItem,
  Button,
  Level,
  LevelItem,
  PageSection,
  Skeleton,
  Tab,
  TabContent,
  Tabs,
  TabsComponent,
  Title,
} from "@patternfly/react-core";
import type { ComponentType } from "react";
import { lazy, Suspense, useRef } from "react";
import {
  Link,
  Redirect,
  Route,
  Switch,
  useHistory,
  useParams,
  useRouteMatch,
} from "react-router-dom";
import { Loading } from "../../../shared";
import { useKafkaInstanceDrawer } from "../KafkaInstanceDrawer";
import { useInstance } from "./support";

const LazyDashboard = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleDashboard")), 1500)
    )
);

const LazyTopics = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleTopics")), 1500)
    )
);

export const KafkaInstanceDetailDemo = () => {
  const { id, tab = "dashboard" } = useParams<{
    id: string;
    tab: string | undefined;
  }>();
  const history = useHistory();
  const match = useRouteMatch({ path: "/streams/:id" });
  const { instance } = useInstance(id);
  const { openDrawer } = useKafkaInstanceDrawer();
  const dashboardRef = useRef<HTMLDivElement>(null);
  const topicsRef = useRef<HTMLDivElement>(null);
  return (
    <>
      <PageSection variant={"light"} type={"breadcrumb"}>
        <Breadcrumb>
          <BreadcrumbItem
            render={() => <Link to={"/streams"}>Kafka instances</Link>}
          />
          <BreadcrumbItem isActive>
            {instance?.name || <Skeleton width={"150px"} />}
          </BreadcrumbItem>
        </Breadcrumb>
      </PageSection>
      <PageSection variant={"light"}>
        <Level>
          <LevelItem>
            <Title headingLevel={"h1"}>
              {instance?.name || <Skeleton width={"150px"} />}
            </Title>
          </LevelItem>
          <LevelItem>
            {instance && (
              <Button onClick={() => openDrawer(instance)}>Open drawer</Button>
            )}
          </LevelItem>
        </Level>
      </PageSection>
      <PageSection variant={"light"} type={"tabs"}>
        <Tabs
          activeKey={tab}
          onSelect={(_, key) => {
            history.push({
              pathname: `${match?.url || ""}/${key}`,
            });
          }}
          component={TabsComponent.nav}
          aria-label="Tabs in the toggled separate content example"
        >
          <Tab
            eventKey={"dashboard"}
            title="Dashboard"
            tabContentId="dashboard-tab"
          />
          <Tab eventKey={"topics"} title="Topics" tabContentId="topics-tab" />
        </Tabs>
      </PageSection>
      <PageSection isFilled={true}>
        <Suspense fallback={<Loading />}>
          <Switch>
            <Route path={`${match?.path || ""}/dashboard`}>
              <TabContent
                eventKey={"dashboard"}
                id="dashboard-tab"
                ref={dashboardRef}
                aria-label="TODO"
              >
                <LazyDashboard />
              </TabContent>
            </Route>
            <Route path={`${match?.path || ""}/topics`}>
              <TabContent
                eventKey={"topics"}
                id="topics-tab"
                ref={topicsRef}
                aria-label="TODO"
              >
                <LazyTopics />
              </TabContent>
            </Route>
            <Redirect
              path={match?.path}
              exact={true}
              to={`${match?.path || ""}/dashboard`}
            />
          </Switch>
        </Suspense>
      </PageSection>
    </>
  );
};

export default KafkaInstanceDetailDemo;
