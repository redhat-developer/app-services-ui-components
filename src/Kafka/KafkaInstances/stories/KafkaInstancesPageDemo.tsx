import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
