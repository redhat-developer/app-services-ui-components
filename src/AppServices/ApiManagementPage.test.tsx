import { composeStories } from "@storybook/testing-react";
import { render, waitForI18n } from "../test-utils";
import * as stories from "./APIManagementPage.stories";

const { APIManagementPage } = composeStories(stories);

describe("ApiManagementPage", () => {
  it("renders", async () => {
    const comp = render(<APIManagementPage />);
    await waitForI18n(comp);
    expect(comp.baseElement).toMatchSnapshot();
  });
});
