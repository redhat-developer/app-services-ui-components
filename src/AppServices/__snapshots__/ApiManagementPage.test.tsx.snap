// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiManagementPage renders 1`] = `
<body>
  <div>
    <section
      class="pf-c-page__main-section pf-m-light appsrv-marketing--banner pf-u-background-color-100"
      style="--appsrv-marketing--banner--before--BackgroundImage: url(dummyimage); --appsrv-marketing--banner--before--BackgroundSize: 478px; --appsrv-marketing--banner--before--BackgroundRepeat: no-repeat; --appsrv-marketing--banner--before--BackgroundPositionY: -99px;"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <h1
          class="pf-c-title pf-m-2xl"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Get started with Red Hat OpenShift API Management
        </h1>
        <p
          class="appsrv-marketing--banner__tagline pf-u-color-200"
          data-ouia-component-id="OUIA-Generated-Text-1"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Add-on service for managed OpenShift
        </p>
        <p
          class=""
          data-ouia-component-id="OUIA-Generated-Text-2"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Based on Red Hat 3scale API management, OpenShift API Management accelerates time to value and reduces the operational cost of delivering API-first, microservices-based applications.
        </p>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-limit-width appsrv-marketing--page-section--marketing"
    >
      <div
        class="pf-c-page__main-body"
      >
        <div
          class="pf-l-grid pf-m-all-6-col-on-lg pf-m-gutter"
        >
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-1"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <div
                  class="pf-c-card__title"
                >
                  <h3
                    class="pf-c-title pf-m-lg"
                    data-ouia-component-id="OUIA-Generated-Title-2"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    Already have OpenShift API Management?
                  </h3>
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              If OpenShift API Management has already been added to your cluster, access the service from the OpenShift web console. Otherwise, use the OpenShift Cluster Manager to add the service to a new or existing managed OpenShift cluster.
            </div>
            <div
              class="pf-c-card__footer"
            >
              <div
                class="pf-l-stack pf-m-gutter"
              >
                <div
                  class="pf-l-stack__item"
                >
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-secondary"
                    data-ouia-component-id="OUIA-Generated-Button-secondary-1"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="cardHaveRHOAM-buttonGoOCM"
                    href="https://cloud.redhat.com/openshift/"
                  >
                    Go to OpenShift Cluster Manager
                  </a>
                </div>
                <div
                  class="pf-l-stack__item"
                >
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-link"
                    data-ouia-component-id="OUIA-Generated-Button-link-1"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="cardHaveRHOAM-linkViewDocs"
                    href="https://access.redhat.com/products/red-hat-openshift-api-management"
                    target="_blank"
                  >
                    View documentation
                     
                    <svg
                      aria-hidden="true"
                      class="pf-u-ml-sm"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </a>
                </div>
              </div>
               
            </div>
          </article>
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-2"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class="pf-c-card__title"
              >
                <h3
                  class="pf-c-title pf-m-lg"
                  data-ouia-component-id="OUIA-Generated-Title-3"
                  data-ouia-component-type="PF4/Title"
                  data-ouia-safe="true"
                >
                  Want to try OpenShift API Management?
                </h3>
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              Explore the many resources available for the OpenShift API Management service, including videos, use cases, and documentation. The 3scale SaaS no-cost trial can help you evaluate the service; the capabilities and experience are similar to OpenShift API Management.
            </div>
            <div
              class="pf-c-card__footer"
            >
              <div
                class="pf-l-stack pf-m-gutter"
              >
                <div
                  class="pf-l-stack__item"
                >
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-secondary"
                    data-ouia-component-id="OUIA-Generated-Button-secondary-2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="cardTryRHOAM-buttonGetStarted"
                    href="https://developers.redhat.com/products/rhoam/getting-started"
                    target="_blank"
                  >
                    Get started with API Management
                    <svg
                      aria-hidden="true"
                      class="pf-u-ml-md"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </a>
                </div>
                <div
                  class="pf-l-stack__item"
                >
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-link"
                    data-ouia-component-id="OUIA-Generated-Button-link-2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="cardTryRHOAM-linkViewDocs"
                    href="https://access.redhat.com/products/red-hat-openshift-api-management"
                    target="_blank"
                  >
                    View documentation
                     
                    <svg
                      aria-hidden="true"
                      class="pf-u-ml-sm"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </a>
                </div>
              </div>
            </div>
          </article>
        </div>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-light pf-m-limit-width appsrv-marketing--page-section--marketing pf-u-background-color-100"
    >
      <div
        class="pf-c-page__main-body"
      >
        <h3
          class="pf-c-title pf-m-xl pf-u-mb-lg"
          data-ouia-component-id="OUIA-Generated-Title-4"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Use Red Hat OpenShift API Management to build an app in five steps
        </h3>
        <div
          class="pf-l-grid pf-m-gutter"
        >
          <div
            class="pf-l-grid__item pf-m-7-col-on-md"
          >
            <article
              class="pf-c-card appsrv-marketing--video"
              data-ouia-component-id="OUIA-Generated-Card-3"
              data-ouia-component-type="PF4/Card"
              data-ouia-safe="true"
              id=""
            >
              <div
                class="appsrv-marketing--video__wrapper"
              >
                <iframe
                  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
                  allowfullscreen=""
                  frameborder="0"
                  src="https://www.youtube.com/embed/NzNgc0f75pc"
                  title="Use Red Hat OpenShift API Management to build an app in five steps"
                />
              </div>
            </article>
          </div>
          <div
            class="pf-l-grid__item pf-m-5-col-on-md"
          >
            <div
              class="pf-c-content"
            >
              <p
                class="pf-u-color-200 pf-u-ml-md"
                data-ouia-component-id="OUIA-Generated-Text-3"
                data-ouia-component-type="PF4/Text"
                data-ouia-safe="true"
                data-pf-content="true"
              >
                In this video, you'll learn how to:
              </p>
              <ul
                class="pf-c-list app-services-ui--icon-list"
              >
                <li
                  class=""
                >
                  Build an application using a Quarkus framework on Red Hat OpenShift Dedicated.
                </li>
                <li
                  class=""
                >
                  Import that application into an OpenShift API Management service.
                </li>
                <li
                  class=""
                >
                  Add an API Key security to protect the API endpoints.
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;
